ci/travis/install-dependencies.sh

```diff
@@ -1,122 +1,173 @@
+  curl -s -L -o "${miniconda_target}" "${miniconda_url}"
+  chmod +x "${miniconda_target}"
+  case "${OSTYPE}" in
+    msys*)
+      miniconda_dir="${miniconda_dir}/bin"  # HACK: Compensate for python.exe being in the installation root on Windows
+      MSYS2_ARG_CONV_EXCL="*" "${miniconda_target}" /S /D="$(cygpath -w -- "${miniconda_dir}")"
+      ;;
+    *)
+      "${miniconda_target}" -b -p "${miniconda_dir}" | grep --line-buffered -v "^\(installing: \|installation finished\.\)"
+      ;;
+  esac
+  { local set_x="${-//[^x]/}"; } 2> /dev/null  # save set -x to suppress noise
+  set +x
+  local source_line='PYTHON3_BIN_PATH=python; PATH="$HOME/miniconda/bin:$PATH"; export PYTHON3_BIN_PATH PATH;'
+  test -f ~/.bashrc && grep -x -q -F "${source_line}" -- ~/.bashrc || echo "${source_line}" >> ~/.bashrc
+  test -z "${set_x}" || set -x  # restore set -x
+  eval "${source_line}"
+  python -m pip install --upgrade --quiet pip
+}
+
+install_nvm() {
+  if [ "${OSTYPE}" = msys ]; then
+      local version="1.1.7" NVM_HOME="${HOME}/.nvm"
+      if [ ! -f "${NVM_HOME}/nvm.sh" ]; then
+        mkdir -p -- "${NVM_HOME}"
+        export NVM_SYMLINK="${PROGRAMFILES}\nodejs"
+        (
+          cd "${NVM_HOME}"
+          local target="./nvm-${version}.zip"
+          curl -s -L -o "${target}" "https://github.com/coreybutler/nvm-windows/releases/download/${version}/nvm-noinstall.zip"
+          unzip -q -- "${target}"
+          rm -f -- "${target}"
+          printf "%s\r\n" "root: $(cygpath -w -- "${NVM_HOME}")" "path: ${NVM_SYMLINK}" > settings.txt
+        )
+        printf "%s\n" "export NVM_HOME=\"$(cygpath -w -- "${NVM_HOME}")\"" 'nvm() { "${NVM_HOME}/nvm.exe" "$@"; }' > "${NVM_HOME}/nvm.sh"
+      fi
+  fi
+}
+
+install_pip() {
+  local python=python
+  if command -v python3 > /dev/null; then
+    python=python3
+  fi
+
+  if "${python}" -m pip --version || "${python}" -m ensurepip; then  # If pip is present, configure it
+    "${python}" -m pip install --upgrade --quiet pip
+
+    # If we're in a CI environment, do some configuration
+    if [ "${TRAVIS-}" = true ] || [ -n "${GITHUB_WORKFLOW-}" ]; then
+      "${python}" -W ignore -m pip config -q --user set global.disable-pip-version-check True
+      "${python}" -W ignore -m pip config -q --user set global.no-color True
+      "${python}" -W ignore -m pip config -q --user set global.progress_bar off
+      "${python}" -W ignore -m pip config -q --user set global.quiet True
+    fi
+  fi
+}
+
+install_node() {
+  if [ "${OSTYPE}" = msys ]; then
+    { echo "WARNING: Skipping running Node.js due to module incompatibilities with Windows"; } 2> /dev/null
+  else
+    # Install the latest version of Node.js in order to build the dashboard.
+    (
+      set +x # suppress set -x since it'll get very noisy here
+      . "${HOME}/.nvm/nvm.sh"
+      nvm install node
+      nvm use --silent node
+      npm config set loglevel warn  # make NPM quieter
+    )
+  fi
+}
+
+run_npm() {
```

Comment:
I can't, because `install-dependencies.sh` is sourced, but `install-ray.sh` isn't.

But I also tried not to move around code in this PR so that we don't have migrations and modifications happening at the same time. I can move around code in a later PR though.

Reply:
Wait sorry, I mixed this up with a different function. I can probably move this one :) thanks!